src/PrestaShopBundle/Controller/Admin/Configure/ShowcaseCardController.php

```
@@ -0,0 +1,92 @@
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Controller\Admin\Configure;
+
+use PrestaShop\PrestaShop\Core\Domain\ShowcaseCard\Command\CloseShowcaseCardCommand;
+use PrestaShop\PrestaShop\Core\Domain\ShowcaseCard\Exception\InvalidShowcaseCardNameException;
+use PrestaShop\PrestaShop\Core\Domain\ShowcaseCard\ValueObject\ShowcaseCard;
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use PrestaShopBundle\Security\Annotation\AdminSecurity;
+use PrestaShopBundle\Security\Annotation\DemoRestricted;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @todo Move this to API
+ */
+class ShowcaseCardController extends FrameworkBundleAdminController
+{
+    /**
+     * Saves the user preference of closing the showcase card.
+     *
+     * This action should be performed via POST, and expects two parameters:
+     * - int $close=1
+     * - string $name Name of the showcase card to close
+     *
+     * @see ShowcaseCard
+     *
+     * @AdminSecurity("is_granted(['create', 'update'], 'CONFIGURE')")
+     * @DemoRestricted(redirectRoute="admin_metas_index")
+     *
+     * @return JsonResponse
+     */
+    public function closeShowcaseCardAction(Request $request)
+    {
+        if ($request->isMethod('post') && $request->request->get('close')) {
```

Prefer to fail fast:
```
if (!$request->isMethod('post') || !$request->request->get('close')) {

        return $this->json(
            [
                'success' => false,
                'message' => '',
            ],
            Response::HTTP_BAD_REQUEST
        );
}
```

---

Fixed